### PR TITLE
Replace optparse with argparse

### PR DESCRIPTION
**Pull request by garberg**
Merge of and some extra fixes to the original pull request to replace optparse with argparse by @dosomder, #573.

### CODE REVIEW

File: nipap/nipap-passwd [hunk 1]
```diff
@@ -13,27 +13,29 @@ from nipap.nipapconfig import NipapConfig, NipapConfigError
...
+    parser.add_argument('-n', '--name', dest='name', type=str, help='set user\'s name to NAME')
+    parser.add_argument('-t', '--trusted', action='store_true', dest='trusted', default=False, help='mark user as trusted')
+    parser.add_argument('-r', '--readonly', action='store_true', dest='readonly', default=False, help='set user to read only')
+    parser.add_argument('-f', '--file', dest='db_file', type=str, help="database file [default: read from config]")
+    parser.add_argument('-c', '--config', dest='config', default='/etc/nipap/nipap.conf', type=str, help=
+            'read configuration from CONFIG [default:/etc/nipap/nipap.conf]')
+    parser.add_argument('--version', action='version', version='nipap-passwd version %s' % nipap.__version__)
+    parser.add_argument('--latest-version', action='store_true',
+            help='check that the sqlite database is of latest version')
+    parser.add_argument('--create-database', action='store_true',
```
Comment (plajjan) [inline nipap/nipap-passwd]: Should this be here? Haven't we covered this action on line 18? Same for latest-version and upgrade-database

File: nipap-cli/nipap [hunk 2]
```diff
@@ -53,13 +53,11 @@ if __name__ == '__main__':
     nullh = NullHandler()
     logger.addHandler(nullh)
 
-    import optparse
-    parser = optparse.OptionParser()
-    parser.add_option("-f", "--force", action="store_true", help="disable interactive prompting of actions")
-    parser.add_option("--show-interpretation", action="store_true", help="show search interpretation")
```
Comment (plajjan) [inline nipap-cli/nipap]: I'm guessing there was a conflict here and show interpretation was accidentally deleted.

**Comment by garberg** [comment]
Yeah, that's right. Fixed!

**Comment by plajjan** [comment]
Ok, I think it looks good now!

**Pull request merged**